src/mdns/mdns_avahi.cpp

```diff
@@ -1167,17 +1167,17 @@ void PublisherAvahi::HostSubscription::HandleResolveResult(AvahiRecordBrowser *
     OTBR_UNUSED_VARIABLE(aFlags);
 
     Ip6Address address;
-    bool       resolved = false;
+    bool       resolved   = false;
+    int        avahiError = AVAHI_OK;
 
     VerifyOrExit(aEvent == AVAHI_BROWSER_NEW);
+    VerifyOrExit(aSize == 16 || aSize == 4, otbrLogErr("unexpected address data length: %u", aSize),
```

Comment:
Can we use `sizeof()` instead of magic numbers here? If not `sizeof()`, then defined constants?